File: UI/system/components/reportBar/dsrc/PrintBar.java
import java.util.HashMap;
import java.util.Map;

import org.dom4j.Element;

import com.justep.ui.system.component.BaseBar;
import com.justep.xbl.runtime.XBLException;

public class PrintBar extends BaseBar {
	private static final String REPORT_PAGE_SETUP_ITEM = "report-page-setup-item";
	private static final String REPORT_PREVIEW_ITEM = "report-preview-item";
	private static final String REPORT_PRINT_ITEM = "report-print-item";

	private String report;

	@SuppressWarnings("rawtypes")
	protected void preProcess(Element bound, Map context) throws XBLException{
		this.report = bound.attributeValue("report");
		this.imgPath = "/UI/system/images/report/";
	}
	
	protected HashMap<String, BarItemDef> getBarItemDefs(){
		HashMap<String, BarItemDef> result = new HashMap<String, BarItemDef>();
		result.put(REPORT_PAGE_SETUP_ITEM, new BarItemDef(REPORT_PAGE_SETUP_ITEM, "print_page_setup.gif", "print_page_setup_g.gif", "call('justep.Context.isReadonlyMode')",
				"justep.xbl('" + report + "').pageSetup();"));
		result.put(REPORT_PREVIEW_ITEM, new BarItemDef(REPORT_PREVIEW_ITEM, "print_preview.gif", "print_preview_g.gif", "call('justep.Context.isReadonlyMode')",
				"justep.xbl('" + report + "').preview();"));
		result.put(REPORT_PRINT_ITEM, new BarItemDef(REPORT_PRINT_ITEM, "print_print.gif", "print_print_g.gif", "call('justep.Context.isReadonlyMode')",
				"justep.xbl('" + report + "').print();"));
		return result;
	}
}

File: UI/system/components/reportBar/designer/printBar.js
justep.design.ReportPrintBar = function(config) {  
	justep.design.ReportPrintBar.superclass.constructor.call(this, config);
};

//依赖justep.design.BaseBar,在bar中实现
justep.extend(justep.design.ReportPrintBar, justep.design.BaseBar, {
	getDefaultItems:function(){
		var basePath =  ComponentConfig[this.componentName].basePath + "/images/";
		return [ ["report-page-setup-item","打印设置",basePath +"print_page_setup.gif",true],
		         ["report-preview-item","打印预览",basePath +"print_preview.gif",true],
				 ["report-print-item","打印",basePath +"print_print.gif",true]
				 ];
	}	
});
 
